frontend/src/__tests__/utils/form-actions.test.ts

```
@@ -97,6 +99,174 @@ describe('form-actions', () => {
+      type: 'application/pdf',
+    });
+    const csv = new File(['file contents'], 'sample.csv', {
+      type: 'text/csv',
+    });
+
+    const response = await createLetterTemplate(
+      createLetterTemplateInput,
+      pdf,
+      csv
+    );
+
+    expect(mockedTemplateClient.createLetterTemplate).toHaveBeenCalledWith(
+      createLetterTemplateInput,
+      'token',
+      pdf,
+      csv
+    );
+
+    expect(response).toEqual(responseData);
+  });
+
+  test('createLetterTemplate accepts empty csv', async () => {
+    const responseData = {
+      templateType: 'LETTER',
+      id: 'new-template-id',
+      templateStatus: 'NOT_YET_SUBMITTED',
+      name: 'template-name',
+      letterType: 'q1',
+      language: 'ar',
+      files: {
+        pdfTemplate: {
+          fileName: 'template.pdf',
+          currentVersion: 'pdf-version',
+          virusScanStatus: 'PENDING',
+        },
+      },
+      createdAt: '2025-01-13T10:19:25.579Z',
+      updatedAt: '2025-01-13T10:19:25.579Z',
+    } satisfies TemplateDto;
+
+    mockedTemplateClient.createLetterTemplate.mockResolvedValueOnce({
+      data: responseData,
+    });
+
+    const createLetterTemplateInput: CreateLetterTemplate = {
+      templateType: 'LETTER',
+      name: 'name',
+      letterType: 'x0',
+      language: 'en',
+    };
+
+    const pdf = new File(['file contents'], 'template.pdf', {
+      type: 'application/pdf',
+    });
+    const csv = new File([], '', {
+      type: 'text/csv',
+    });
+
+    const response = await createLetterTemplate(
+      createLetterTemplateInput,
+      pdf,
+      csv
+    );
+
+    expect(mockedTemplateClient.createLetterTemplate).toHaveBeenCalledWith(
+      createLetterTemplateInput,
+      'token',
+      pdf,
+      undefined
+    );
+
+    expect(response).toEqual(responseData);
+  });
+
+  test('createLetterTemplate - should thrown error when saving unexpectedly fails', async () => {
```

Just a small typo - should throw an error...